### PR TITLE
Support oauth2login configuration for GrantedAuthoritiesMapper using @Bean

### PR DESCRIPTION
**Pull request by kazuki43zoo**
I've fixed gh-4880.
Please review this.

### CODE REVIEW

**Comment by jgrandja** [comment]
Thanks for the PR @kazuki43zoo.

I need to address #4909 before this PR can be considered for merging.
I'll address this soon.

**Comment by kazuki43zoo** [comment]
I've rebased.

File: config/spring-security-config.gradle [hunk 1]
```diff
@@ -66,6 +66,7 @@ dependencies {
 
 	testRuntime 'cglib:cglib-nodep'
 	testRuntime 'org.hsqldb:hsqldb'
+	testRuntime 'com.fasterxml.jackson.core:jackson-databind'
```
Comment (jgrandja) [inline config/spring-security-config.gradle]: Do we need to include this dep?

File: config/src/test/groovy/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2LoginConfigurerTests.groovy [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import org.springframework.security.oauth2.client.oidc.authentication.OidcAuthorizationCodeAuthenticationProvider
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository
+import org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository
+import org.springframework.security.oauth2.client.web.OAuth2LoginAuthenticationFilter
+
+/**
+ * test for OAuth2LoginConfigurer.
+ * @author Kazuki Shimizu
+ */
+class OAuth2LoginConfigurerTests extends BaseSpringSpec {
```
Comment (jgrandja) [inline config/src/test/groovy/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2LoginConfigurerTests.groovy]: Can you please convert this to JUnit. We're in the process of converting all our tests to JUnit as we're discontinuing Groovy.

File: config/spring-security-config.gradle (hunk 1 above)
Comment (kazuki43zoo) [reply config/spring-security-config.gradle]: yes.
If it not include, the following exception occurred.

```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'springSecurityFilterChain' defined in org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [javax.servlet.Filter]: Factory method 'springSecurityFilterChain' threw exception; nested exception is java.lang.NoClassDefFoundError: com/fasterxml/jackson/core/util/DefaultPrettyPrinter$Indenter

	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:587)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1250)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1099)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:545)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:502)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:312)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:228)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:310)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:304)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:758)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:868)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:549)
	at org.springframework.security.config.annotation.BaseSpringSpec.loadConfig(BaseSpringSpec.groovy:96)
	at org.springframework.security.config.annotation.web.configurers.oauth2.client.OAuth2LoginConfigurerTests.oauth2Login default(OAuth2LoginConfigurerTests.groovy:42)
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [javax.servlet.Filter]: Factory method 'springSecurityFilterChain' threw exception; nested exception is java.lang.NoClassDefFoundError: com/fasterxml/jackson/core/util/DefaultPrettyPrinter$Indenter
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:579)
	... 15 more
Caused by: java.lang.NoClassDefFoundError: com/fasterxml/jackson/core/util/DefaultPrettyPrinter$Indenter
	at org.springframework.security.oauth2.client.userinfo.NimbusUserInfoResponseClient.<init>(NimbusUserInfoResponseClient.java:51)
	at org.springframework.security.oauth2.client.userinfo.DefaultOAuth2UserService.<init>(DefaultOAuth2UserService.java:51)
	at org.springframework.security.config.annotation.web.configurers.oauth2.client.OAuth2LoginConfigurer.init(OAuth2LoginConfigurer.java:241)
...
```

File: config/src/test/groovy/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2LoginConfigurerTests.groovy (hunk 2 above)
Comment (kazuki43zoo) [reply config/src/test/groovy/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2LoginConfigurerTests.groovy]: of course!

Comment (kazuki43zoo) [reply config/src/test/groovy/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2LoginConfigurerTests.groovy]: I've fixed via b876aada8c326d6cd8d59e8db8a075fec631ac86.

**Comment by jgrandja** [comment]
Thanks for the PR @kazuki43zoo! I've applied some minor polish and added doc to the reference. This is now in master.

**Pull request closed without merging**